src/math/Polynomial.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <vector>
+
+namespace NuTo {
+namespace Math
+{
+
+class Polynomial
```

Comment:
IMO the name `Polynomial` is misleading, it's more like `Legendre`, right?

And please provide a unit test, e.g. a comparison to wikipedia values of the gauss-lobatto points.